### ISSUE TITLE
Incompatibility with Tslint 5

### ISSUE THREAD

**Issue by philipmw** [Microsoft/tslint-microsoft-contrib#363]
Given these dependencies:

    "devDependencies": {
      "typescript": "latest",
      "tslint": "latest",
      "tslint-microsoft-contrib": "latest"
    }

npm installs `tslint` 5.1.0, and I have this problem:

    $ ./node_modules/tslint/bin/tslint --typecheck -t verbose src/API.ts
    /Path/To/Repo/node_modules/tslint/lib/runner.js:116
                throw error;
                ^

    TypeError: Cannot read property 'prototype' of undefined
        at __extends (/Path/To/Repo/node_modules/tslint-microsoft-contrib/noSuspiciousCommentRule.js:5:68)
        at /Path/To/Repo/node_modules/tslint-microsoft-contrib/noSuspiciousCommentRule.js:38:5
        at Object.<anonymous> (/Path/To/Repo/node_modules/tslint-microsoft-contrib/noSuspiciousCommentRule.js:77:2)
        at Module._compile (module.js:571:32)
        at Object.Module._extensions..js (module.js:580:10)
        at Module.load (module.js:488:32)
        at tryModuleLoad (module.js:447:12)
        at Function.Module._load (module.js:439:3)
        at Module.require (module.js:498:17)
        at require (internal/module.js:20:19)

I get the same problem with tslint 5.0.0.

But when I downgrade `tslint` to `<5`, like so:

    "devDependencies": {
      "typescript": "latest",
      "tslint": "<5",
      "tslint-microsoft-contrib": "latest"
    }

Then everything works as expected.

I am not certain that you officially support tslint 5.x, but the README suggests that any version >= 4 is supported.

**Comment by jmmk** [comment]
Seems to be due to https://github.com/palantir/tslint/pull/2370, which removes `SkippableTokenAwareRuleWalker`

**Comment by HamletDRC** [comment]
Thanks for the report. We are trying to fix this under the issue #354 , so I am going to close this as a duplicate.

**Comment by userpixel** [comment]
I got a similar issue:

```
[Info  - 10:15:43 PM] vscode-tslint: 'Cannot read property 'prototype' of undefined' while validating: /home/user/learn/typescript/src/index.ts stacktrace: TypeError: Cannot read property 'prototype' of undefined
    at __extends (/home/user/learn/typescript/node_modules/tslint-microsoft-contrib/noSuspiciousCommentRule.js:5:68)
    at /home/user/learn/typescript/node_modules/tslint-microsoft-contrib/noSuspiciousCommentRule.js:38:5
    at Object.<anonymous> (/home/user/learn/typescript/node_modules/tslint-microsoft-contrib/noSuspiciousCommentRule.js:77:2)
    at Module._compile (module.js:556:32)
    at Object.Module._extensions..js (module.js:565:10)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
```

# Versions

```
    "@types/node": "^7.0.12",
    "tslib": "^1.6.0",
    "tslint": "^5.1.0",
    "tslint-microsoft-contrib": "^4.0.1",
    "typedoc": "^0.5.10",
    "typescript": "^2.2.2"
```
# TSLint settings

```
// Place your settings in this file to overwrite default and user settings.
{
    "tslint.enable": true,
    "tslint.jsEnable": false,
    "tslint.rulesDirectory": "node_modules/tslint-microsoft-contrib",
    "tslint.exclude": "node_modules/**/*",
    "tslint.autoFixOnSave": true
}
```
